src/plugins/navigation_embeddable/public/components/editor/navigation_embeddable_link_destination.tsx

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useState } from 'react';
+
+import { DashboardContainer } from '@kbn/dashboard-plugin/public/dashboard_container';
+
+import {
+  NavigationLinkType,
+  EXTERNAL_LINK_TYPE,
+  DASHBOARD_LINK_TYPE,
+} from '../../../common/content_management';
+import { NavigationEmbeddableUnorderedLink } from '../../editor/open_link_editor_flyout';
+import { ExternalLinkDestinationPicker } from '../external_link/external_link_destination_picker';
+import { DashboardLinkDestinationPicker } from '../dashboard_link/dashboard_link_destination_picker';
+import { NavEmbeddableStrings } from '../navigation_embeddable_strings';
+import { EuiFormRow } from '@elastic/eui';
+
+export const NavigationEmbeddableLinkDestination = ({
```

Comment:
I separated this out into a new component for two reasons:
1. It cleans up the logic of `NavigationEmbeddableLinkEditor` a little bit
2. It makes it easier to remember selections even when the link types change, like so:

| Before | After |
|--------|--------|
| ![Aug-30-2023 17-31-14](https://github.com/elastic/kibana/assets/8698078/021d71db-1da0-408a-976b-33dff3056a33) | ![Aug-30-2023 17-30-41](https://github.com/elastic/kibana/assets/8698078/a669ccab-32ac-43cf-8d55-1a55fc8ae47c) |